packages/aragon-cli/src/commands/apm_cmds/publish.js

```diff
@@ -107,8 +107,50 @@ exports.builder = function(yargs) {
     })
 }
 
+const getMajor = version => version.split('.')[0]
+
+function getFunctionsAbi(functions, abi) {
+  functions.forEach(f => {
+    f.abi = abi.find(method => method.name === f.sig.match(/(.*)\(/)[1])
+  })
+}
+
+async function getDeprecatedFunctions(apm, artifact) {
+  let deprecated = []
+  try {
+    let deprecatedOnVersion = []
+    const deprecatedFunctionsSig = []
+    const versions = await apm.getAllVersions(artifact.appName)
+    let lastMajor = -1
+    versions.forEach(async repo => {
+      // iterate on major versions
+      if (getMajor(repo.version) !== lastMajor) {
+        lastMajor = getMajor(repo.version)
+        repo.functions.forEach(f => {
+          if (
+            !artifact.functions.some(obj => obj.sig === f.sig) &&
+            !deprecatedFunctionsSig.includes(f.sig)
+          ) {
+            deprecatedOnVersion.push(f)
+            deprecatedFunctionsSig.push(f.sig)
+          }
+        })
+        if (deprecatedOnVersion.length) {
+          deprecated[`${lastMajor}.0.0`] = deprecatedOnVersion
```

Comment:
@sohkai or @0x6431346e can you help with this line. I been struggling with JSON and how to add the array of deprecated functions. I tested the content of deprecatedOnVersionand is correct in this line, but is never assigned to deprecated (deprecated remain always [ ]). I also tested a simpler example on the console of the browser and it work like this 😔

Reply:
@0xGabi Unfortunately you can't assigned named keys to an array in JS, you'll have to use an object (`{}`) instead.

We can iterate over an object quite easily using helpers like `Object.entries()`, `for/of`, and etc. so don't worry about making `deprecated` an array :).